docs/adr/adr_improved_frontend_dx.md

```diff
@@ -0,0 +1,55 @@
+# ADR: Improved Frontend Developer Experience
+Decision Deadline: As soon as possible
+
+Author: [Felipe](https://github.com/fhenrich33)
+
+## Status
+
+Proposed, Implementing
+
+## Context
+Frontend development experience in its current form is quite lacking. Imposes a lot of artificial difficulties, tooling doesn't work as expected, and adds friction to our FE contributors.
+
+We want to make FE development better.
+
+## Decision drivers
+
+1. We must make FE onboarding and development seamless.
+2. Implementation must play well with the current FE setup (React/Vite).
+3. Implementation must work well with our CI/CD pipeline.
+4. We must NOT interfere with BE tooling and development.
+
+## Considered options
+
+The only difference between proposed options is how to manage the monorepo, otherwise, there's a lot of overlap on proposed changes:
+- **[pnpm](https://pnpm.io/):** The "new" unwritten standard for package management--faster than both `npm` and `yarn`, and more forgiving on how it manages and caches `node_modules`. Will replace `yarn`.
+- **[husky](https://typicode.github.io/husky/) and [lint-staged](https://github.com/lint-staged/lint-staged):** Improved git hook configuration, also an unwritten standard in the FE ecosystem. Will replace our manual git hook setup.
```

Comment:
Does "unwritten standard" here mean "the de facto standard"?

Reply:
I would not say a de facto standard since we don't even have a standard library for Javascript. But since it's very commonplace, that's why I picked these words specifically.

Reply:
What's the manual git hook setup?

Reply:
Sorry, poor choice of words on my end: I meant about the hand-rolled scripts like the one to invoke Yarn.

Reply:
That's not ideal indeed 😅